### ISSUE TITLE
Incidents for the "JBoss API reference " issue not found in neither application source nor dependencies

### ISSUE THREAD

**Issue by rromannissen** [konveyor/analyzer-lsp#329]
### Environment

- Fedora 36
- minikube v1.31.2 (--driver=kvm2 --memory=10g --cpus=3)
- Konveyor 0.3 alpha-2 with additional images:
  - ui_image_fqin: quay.io/ibolton/tackle2-ui:demo
  - hub_image_fqin: quay.io/jortel/tackle2-hub:demo
  - analyzer_fqin: quay.io/jortel/tackle2-addon-analyzer:demo
- [tackle-testapp](https://github.com/konveyor/tackle-testapp) application
- Analysis configuration:
  - Source + Dependencies
  - eap6
  - Application and internal dependencies only
  - No custom rules
  - No additional advanced options

### Description

The _JBoss API reference_ issue is not generating incidents when analyzing the tackle-testapp application, neither in its source code nor the associated dependencies, when it should be found in both.

### How to reproduce

- Analyze tackle-test app with the configuration available in the Environment section.
- Go to the issues view, Single application and filter by the tackle-testapp application.

### Expected result

The _JBoss API reference_ issue should appear in the list of issues for the application. Windup analysis reports 3671 incidents in the dependencies and 6 incidents in the application source:

![Screenshot from 2023-09-12 11-54-28](https://github.com/konveyor/analyzer-lsp/assets/15966456/db9469ee-7bf4-4a84-970f-5b45b9e7a116)

![Screenshot from 2023-09-12 11-54-38](https://github.com/konveyor/analyzer-lsp/assets/15966456/bf08d49d-162b-4c44-8e81-b2234bf4de19)

### Actual result

The _JBoss API reference_ issue doesn't appear in the list of issues for the tackle-testapp application.

**Comment by pranavgaikwad** [comment]
@shawn-hurley I think this is related to package search thing. Here's the rule that's expected to be triggered:

```
- category: potential
  customVariables:
  - name: type
    nameOfCaptureGroup: type
    pattern: org.(?P<subpackage>(jboss|jbpm).)?(?P<package-remainder>([a-z]+\.)*)?(?P<type>[^.()]+)?
  - name: subpackage
    nameOfCaptureGroup: subpackage
    pattern: org.(?P<subpackage>(jboss|jbpm).)?(?P<package-remainder>([a-z]+\.)*)?(?P<type>[^.()]+)?
  - name: package-remainder
    nameOfCaptureGroup: package-remainder
    pattern: org.(?P<subpackage>(jboss|jbpm).)?(?P<package-remainder>([a-z]+\.)*)?(?P<type>[^.()]+)?
  description: |
    JBoss API reference
  effort: 0
  labels:
  - konveyor.io/source=java
  - konveyor.io/source=javaee
  - konveyor.io/source=soa
  - konveyor.io/target=eap6
  - phase=PostMigrationRulesPhase
  - catchall
  - jboss
  links: []
  message: '`org.{{subpackage}}.{{package-remainder}}{{type}}` reference found. No
    specific details available.'
  ruleID: generic-catchall-00700
  when:
    java.referenced:
      pattern: org.(jboss|jbpm).([a-z]+.)*[^.()]+
```

**Comment by shawn-hurley** [comment]
I have a feeling you are correct, but I see a different problem. 

Assuming that the maven index is working, I assume that the org.jboss and org.jbpm are open-source deps and may be filtered out here. 

> Application and internal dependencies only

I could be wrong, but does Windup have some special logic?

**Comment by pranavgaikwad** [comment]
@shawn-hurley I see two screenshots, first, equivalent to "full" scope + dep filter, second, equivalent to "full" scope and no filter of deps. @rromannissen In the windup screenshots here, I am guessing you ran analysis twice with different scopes, is that right?

**Comment by shawn-hurley** [comment]
If that is the case, then the mismatch in a number of issues does make sense, or am I missing something?

**Comment by rromannissen** [comment]
Nope, the Windup report treats the application source and the dependencies as two different "applications" with different issue lists. This is a single report that got generated when I analyzed the application in "Source+Dependencies" with "Application and internal dependencies only".

**Comment by pranavgaikwad** [comment]
@rromannissen That is weird. Regardless of how windup does it, I think what should happen here is that: since you selected "source + dependencies", the analysis scope should be set to "full" which will enable analyzer to look into incidents in deps. But since you selected "application and internal deps only", all the incidents in deps "org.jboss" should get filtered. So you should see 6 matches for this rule in config-utils, but not those other 3000 matches. Although in this case, it seems that the rule is not matching at all.

**Comment by shawn-hurley** [comment]
Wait, it is finding the ones in the source code, right?

**Comment by pranavgaikwad** [comment]
No, it's not as per description
> The JBoss API reference issue is not generating incidents when analyzing the tackle-testapp application, neither in its source code nor the associated dependencies, when it should be found in both.

**Comment by rromannissen** [comment]
> But since you selected "application and internal deps only", all the incidents in deps "org.jboss" should get filtered. So you should see 6 matches for this rule in config-utils, but not those other 3000 matches.

It should be like that if the image I have wasn't affected by #320, but since it is, I guess it should display both. I can confirm neither of them are being displayed.
